### ISSUE TITLE
Add plaintext export

### ISSUE THREAD

**Issue by alexmshepherd** [owdex/owdex#11]
In order to [reindex](https://solr.apache.org/guide/solr/latest/indexing-guide/reindexing.html) it's necessary to have a standardised ingest process. This could take the form of a CSV with columns `url`, `title` and `submitter`, since other attributes are non-static.

**Comment by 4censord** [comment]
Consider using [jsonlines](https://jsonlines.org/). 

They are defined and standardized slightly better than CSV, and can easily parse in parallel, as they don't need a header.

**Comment by alexmshepherd** [comment]
> Consider using [jsonlines](https://jsonlines.org/).

I like this idea a lot. Thanks for putting it forward :)

**Comment by 4censord** [comment]
This is also be a contender for background execution like #12
We could also consider making this a script to be run independently by something like cron.

Also, we should think about the export format.
Huge plain text files get difficult to manage at huge scale, so some kind of splitting might be required.

**Comment by alexmshepherd** [comment]
well, using jsonlines, splitting would just be a matter of moving to a new file after _x_ entries, which would be pretty nice.

**Comment by alexmshepherd** [comment]
i agree that regularly running this as a background task and saving the results would be a better backup solution than backing up the entire database.

**Comment by 4censord** [comment]
> well, using jsonlines, splitting would just be a matter of moving to a new file after x entries, which would be pretty nice.

We probably also want some kind of index of files, and file hashes for consistency.

Also, maybe having object storage as the primary backup target might be nice.

**Comment by alexmshepherd** [comment]
what about an archive like .zip with a number of jsonlines files and a "manifest" with hashes?

the reason i think storing this as the backup target would be preferable is that storing the solr-indexed objects would likely be somewhat more storage-heavy. with the export files it would still be possible to bring all the indexed pages back in, just trading processing power in the event of a failure for storage requirements.

**Comment by 4censord** [comment]
I do agree that this also solves the need for backups very nicely.
At a scale where uptime is a serious concern, one would be running a highly available solr cluster anyways.

Zip files are alright for smaller datasets, but if we assume growth to terabytes, or 10s of terabytes they get unwieldy as well.

I was thinking of something like a folder structure:

```shell
.
├── aa
│   ├── BV1n8AVWGm8b.jsonlines.zstd
│   ├── J1tsyn1oWNZb.jsonlines.zstd
│   └── n09F1aMELAt1.jsonlines.zstd
├── ab
│   ├── jH0w7CfTAbRA.jsonlines.zstd
│   ├── l2BPQLQqVSuq.jsonlines.zstd
│   └── Vq1nAzX26YRl.jsonlines.zstd
└── index.json
```

with the index.json looking like this:
```json
{
    "files": [
        {
            "checksums": {
                "sha256": "f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2"
            },
            "filesize": 10564312,
            "fragment_version": 1,
            "path": "aa/BV1n8AVWGm8b.jsonlines.zstd",
            "records": 100000,
            "compression": "zstd"
        },
        {...}
    ],
    "index_version": 1,
    "records": 200000
}
```

**Comment by alexmshepherd** [comment]
This is beginning to be addressed with 963e5da

**Comment by alexmshepherd** [comment]
This has been moved to [owdex/ixport](https://github.com/owdex/ixport)
